plugins/single_plugins/scale.cpp

```diff
@@ -993,16 +1006,18 @@ class wayfire_scale : public wf::plugin_interface_t
         {
             auto view = get_signaled_view(data);
 
-            auto it = scale_data.find(view->parent);
-            if (it != scale_data.end())
+            if (!scale_view(view) && (view->role != wf::VIEW_ROLE_TOPLEVEL))
             {
-                layout_slots(get_views());
-
                 return;
             }
 
-            if (!scale_view(view) && (view->role != wf::VIEW_ROLE_TOPLEVEL))
+            view->connect_signal("geometry-changed", &view_geometry_changed);
+
+            auto it = scale_data.find(view->parent);
+            if (it != scale_data.end())
             {
+                layout_slots(get_views());
+
```

Comment:
here. Why did you change this code? Also I don't really want the `geometry-changed` handler moved out of add_transformer function.

Reply:
The problem is that layout_slots() needs to know if it sees a view for the first time or not, since that allows it to correctly place a child view in a correct location.

In the current version, this is handled by checking the result of add_transformer(), i.e. add_transformer() is only called from within layout_slots() and if it returns true, then it is a new view that was not handled before.

At the same time, the geometry-changed signal needs to be connected earlier, since typically that is what triggers the layout_slots() call that will set up the transformer for a newly appeared child view. For this reason, the geometry-changed signal needs to be set up the first time a view is seen.

There could be other ways to keep track if a view was seen or is a potential new child view. But so far, I could not think of any way where it is clearly signaled that a new child view has been added.

The problem is also that the first time a view is attached, the parent relationship is not set up yet, at least in the cases tested view->parent is null here.

Reply:
Now, I'm not sure what was my reason of swapping the two conditions. Is there a case where the first condition (in the original code, i.e. looking up the parent in the map) is true, while the second condition is false?

Reply:
I realized that my original intention was that the "geometry-changed" signal should be connected in all cases except when the second condition (checking if view is a scale_view) is true. Now I'm less sure if this is necessary; I've added an update that just replaces the add_transformer() call. I'm concerned this could result in the "geometry-changed" signal not being connected if a new view is directly handled by the previous cases. 

I'm wondering if it is OK to connect the same signal multiple times? Does that mean we have to disconnect it the same number of times it was connected?